diff --git a/backends/imgui_impl_vulkan.cpp b/backends/imgui_impl_vulkan.cpp
index 848615ed..64203da1 100644
--- a/backends/imgui_impl_vulkan.cpp
+++ b/backends/imgui_impl_vulkan.cpp
@@ -1218,20 +1218,25 @@ static VkPipeline ImGui_ImplVulkan_CreatePipeline(VkDevice device, const VkAlloc
     VkPipelineCreateFlags2CreateInfo ci_heap_flags{};
     ci_heap_flags.sType = VK_STRUCTURE_TYPE_PIPELINE_CREATE_FLAGS_2_CREATE_INFO;
     ci_heap_flags.flags = VK_PIPELINE_CREATE_2_DESCRIPTOR_HEAP_BIT_EXT;
-    if(bd->VulkanInitInfo.DescriptorHeapInfo)
-    {
-        create_info.pNext = &ci_heap_flags;
-    }
 
 #ifdef IMGUI_IMPL_VULKAN_HAS_DYNAMIC_RENDERING
+    VkPipelineRenderingCreateInfo pipeline_rendering_create_info = {};
     if (bd->VulkanInitInfo.UseDynamicRendering)
     {
         IM_ASSERT(info->PipelineRenderingCreateInfo.sType == VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO_KHR && "PipelineRenderingCreateInfo::sType must be VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO_KHR");
         IM_ASSERT(info->PipelineRenderingCreateInfo.pNext == nullptr && "PipelineRenderingCreateInfo::pNext must be nullptr");
-        create_info.pNext = &info->PipelineRenderingCreateInfo;
+        pipeline_rendering_create_info = info->PipelineRenderingCreateInfo;
+        if (bd->VulkanInitInfo.DescriptorHeapInfo)
+            pipeline_rendering_create_info.pNext = &ci_heap_flags;
+        create_info.pNext = &pipeline_rendering_create_info;
         create_info.renderPass = VK_NULL_HANDLE; // Just make sure it's actually nullptr.
     }
+    else
 #endif
+    if (bd->VulkanInitInfo.DescriptorHeapInfo)
+    {
+        create_info.pNext = &ci_heap_flags;
+    }
     VkPipeline pipeline;
     VkResult err = vkCreateGraphicsPipelines(device, pipelineCache, 1, &create_info, allocator, &pipeline);
     check_vk_result(err);
